### ISSUE TITLE
0 test accuracy and loss goes to nan

### ISSUE THREAD

**Issue by ZiweiXU** [phanideepgampa/stacked-capsule-networks#7]
Hi, thank you for your effort in implementing this!

I tried to invoke your code directly by `python main.py` and run it several times, however, every time it produced 0 test accuracy and loss eventually went to nan after ~200 epochs. 

Please see the log below:

```
2020-04-29 06:23:53,464 DEBUG      epoch 199test-accuracy: tensor(0, device='cuda:0')
2020-04-29 08:16:06,942 DEBUG      epoch 249test-accuracy: tensor(0, device='cuda:0')
2020-04-29 10:08:19,865 DEBUG      epoch 299test-accuracy: tensor(0, device='cuda:0')
```
```
2020-04-29 05:33:43,878 INFO       ==>>> epoch: 177, batch index: 400, train loss: nan
2020-04-29 05:33:58,126 INFO       ==>>> epoch: 177, batch index: 450, train loss: nan
2020-04-29 05:34:03,515 INFO       ==>>> epoch: 177, batch index: 469, train loss: nan
```

Could you advise on this?

**Comment by Add2seconds** [comment]
I also had the problem that the test accuracy was zero, but my train loss dropped from 8000 to 1000 in few iteration, and then it only fluctuated slightly
I hope the author can give a solution

**Comment by FrostbiteXSW** [comment]
This happened to me as well. This nan-loss problem may caused by exploding gradient. By the way, I wonder if this problem is related to pytorch version? I'm using pytorch 1.5 with cuda 9.2.

**Comment by chaturangalagama** [comment]
same issue occurred to me also. I tried running with different versions of pytorch. It didn't work

**Comment by Stark-Chang** [comment]
This happened to me as well.I use the default settings and just run 'python main.py'.At the beginning,The train loss dropped from 1000 to 800 and then became nan in epoch 69. The accuracy is 0.1135 in epoch 49.I hope the author can give a solution

**Comment by HelloWXJ1024** [comment]
m

> This happened to me as well.I use the default settings and just run 'python main.py'.At the beginning,The train loss dropped from 1000 to 800 and then became nan in epoch 69. The accuracy is 0.1135 in epoch 49.I hope the author can give a solution

me too.

And have you solved the problem?

**Comment by HelloWXJ1024** [comment]
> This happened to me as well.I use the default settings and just run 'python main.py'.At the beginning,The train loss dropped from 1000 to 800 and then became nan in epoch 69. The accuracy is 0.1135 in epoch 49.I hope the author can give a solution

I meet the same problem, have you solved it? thx

**Comment by keneathhui** [comment]
I meet the same problem, have you solved it?
